src/main/java/com/algolia/search/models/apikeys/AddApiKeyResponse.java

```diff
@@ -0,0 +1,53 @@
+package com.algolia.search.models.apikeys;
+
+import com.algolia.search.exceptions.AlgoliaApiException;
+import com.algolia.search.models.WaitableResponse;
+import java.io.Serializable;
+import java.time.OffsetDateTime;
+import java.util.function.Function;
+
+public class AddApiKeyResponse implements WaitableResponse, Serializable {
+
+  public String getKey() {
+    return key;
+  }
+
+  public void setKey(String key) {
+    this.key = key;
+  }
+
+  public void setGetApiKeyFunction(Function<String, ApiKey> getApiKeyConsumer) {
+    this.getApiKeyFunction = getApiKeyConsumer;
+  }
+
+  public OffsetDateTime getCreatedAt() {
+    return createdAt;
+  }
+
+  public void setCreatedAt(OffsetDateTime createdAt) {
+    this.createdAt = createdAt;
+  }
+
+  private Function<String, ApiKey> getApiKeyFunction;
+  private String key;
+  private OffsetDateTime createdAt;
+
+  @Override
+  public void waitTask() {
+    // Loop until the key is created on the server => error code != 404
+    while (true) {
+      try {
+        getApiKeyFunction.apply(getKey());
+      } catch (AlgoliaApiException ex) {
+        if (ex.getHttpErrorCode() == 404) {
+          try {
+            Thread.sleep(1001);
+          } catch (InterruptedException ignored) {
+          }
+          continue;
```

Comment:
Isn't it a problem to ignore a `InterruptedException` and simply `continue`? :thinking: 

- https://rules.sonarsource.com/java/tag/multi-threading/RSPEC-2142
- https://stackoverflow.com/questions/28544867/is-it-ok-to-ignore-interruptedexception-if-nobody-calls-interrupt/28545020#28545020

Reply:
👍 Good catch I'll have a look at it.

Reply:
Fixed with 049d954 thanks for catching it